mcs/tools/upload-to-sentry/Makefile

```diff
@@ -0,0 +1,41 @@
+thisdir = tools/upload-to-sentry
+SUBDIRS =
+include ../../build/rules.make
+
+PROGRAM = upload-to-sentry.exe
+LIB_REFS = \
+	System \
+	System.Core \
+	System.Data \
+	System.Runtime.Serialization \
+	System.Xml \
+	System.Xml.Linq \
+	Mono.Posix
+NO_INSTALL = yes
+
+SENTRY_ROOT := $(shell dirname "$(topdir)/../")
+# These come from the environment
+SENTRY_ENV_OPTIONS = MONO_SENTRY_ROOT="$(SENTRY_ROOT)" MONO_SENTRY_OS="$(MONO_SENTRY_OS)" MONO_SENTRY_URL="$(MONO_SENTRY_URL)" 
```

Comment:
I don't understand why you need most of this, the environment variables should already be there?

And for `MONO_SENTRY_ROOT/SENTRY_ROOT ` it might be better to move this out of the Makefile and pass it in from wherever we're calling this script. E.g. remove reading `MONO_SENTRY_ROOT` from the exe and instead make it a command line parameter:

```
...../upload-to-sentry.exe $(MONO_SENTRY_ROOT)
```

and in run-upload-sentry.sh:

```
make -C mcs/tools/upload-to-sentry upload-crashes MONO_SENTRY_ROOT="$MONO_REPO_ROOT"
```